tests/app/dao/test_annual_billing_dao.py

```diff
@@ -91,6 +90,17 @@ def test_set_default_free_allowance_for_service(notify_db_session, org_type, yea
     assert annual_billing[0].free_sms_fragment_limit == expected_default
 
 
+def test_set_default_free_allowance_for_service_fails_if_no_historical_default(notify_db_session):
+    service = create_service(organisation_type="central")
+
+    with pytest.raises(ValueError) as e:
+        set_default_free_allowance_for_service(service=service, year_start=2000)
```

Comment:
Answering your commit message here:

Right now, if a user goes to `https://www.notifications.service.gov.uk/services/<service_name>/usage?year=2018` and their service existed then, they would see a valid result. 

This will break after those changes go in.
The user would have to manipulate their url to get there though, so maybe not worth pulling our hair out over :). But we could also avoid that by defaulting to 2020 allowance if year is older than 2020, since our allowances didn't change before 2020 as far as I'm aware.

Reply:
> Right now, if a user goes to https://www.notifications.service.gov.uk/services/<service_name>/usage?year=2018 and their service existed then, they would see a valid result.

I don't think it _should_ break if the service did already exist in 2018, as they should have an entry in AnnualBilling already, which means they won't need to query the DefaultAnnualAllowance table.

However, I think you make a point that if a service which _didn't_ exist in 2018 hit the same URL, they'll currently see a valid page whereas after this change they'll get a 500.

I'll fix that.

Reply:
Fixed in 3843d44dad0c3a2ba973498a0c1a5780ff7a3702 (which I'll squash before merging)

Reply:
Also made me spot an edge case that would let users 'lock in' allowances for future years, which I've closed in 0046c5d53ff9cb68e06dfdf782eae52737d663e0